src/main/java/ru/mts/Animals/AbstractAnimal.java

```diff
@@ -40,4 +45,48 @@ public String getName() {
     public String getCharacter() {
         return character;
     }
+
+    @Override
+    public LocalDate getBirthDate() {
+        return birthDate;
+    }
+
+    /**
+     * <b>getFormatDate</b>
+     * @param format формат строки
+     * @return дату в формате format
+     */
+    public String getFormatDate(String format)
+    {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern(format);
+        return formatter.format(birthDate);
+    }
+
+    /**
+     * <b>equals</b>
+     * @param obj формат строки
+     * @return true или false в зависимости от равенства объектов
+     */
+    @Override
+    public boolean equals(Object obj) { // будут равны если равны имена, даты рождения и порода
+        if(this == obj) return true;
+
+        if(obj == null || getClass() != obj.getClass()) return false;
+
+        return (name.equals(((AbstractAnimal) obj).name) &&
```

Comment:
```suggestion
        var that = ((AbstractAnimal) obj);
        return Objects.equals(name, that.name)
                && Objects.equals(birthDate, that.birthDate)
                && Objects.equals(breed, that.breed);
```

- вынос переменной более нагляден
- Objects.equals() не упадёт если , скажем, первый name == null